contracts/Aavegotchi/ForgeDiamond/facets/ForgeVRFFacet.sol

```diff
@@ -0,0 +1,249 @@
+import "../libraries/LibAppStorage.sol";
+
+import {LibMeta} from "../../../shared/libraries/LibMeta.sol";
+import {ILink} from "../../interfaces/ILink.sol";
+import {ForgeFacet} from "./ForgeFacet.sol";
+import {ForgeTokenFacet} from "./ForgeTokenFacet.sol";
+import {LibSvg} from "../../libraries/LibSvg.sol";
+
+contract ForgeVRFFacet is Modifiers {
+    event VrfResponse(address user, uint256 randomNumber, bytes32 requestId, uint256 blockNumber);
+    event GeodeWin(address user, uint256 itemId, uint256 geodeTokenId, bytes32 requestId, uint256 blockNumber);
+    event GeodeEmpty(address user, uint256 geodeTokenId, bytes32 requestId, uint256 blockNumber);
+    event GeodeRefunded(address user, uint256 geodeTokenId, bytes32 requestId, uint256 blockNumber);
+
+    function forgeTokenFacet() internal view returns (ForgeTokenFacet facet) {
+        facet = ForgeTokenFacet(address(this));
+    }
+
+    function forgeFacet() internal view returns (ForgeFacet facet) {
+        facet = ForgeFacet(address(this));
+    }
+
+    // Maximum number of geodes that can be opened in one call.
+    uint256 constant MAX_VRF = 15;
+
+    function linkBalance() external view returns (uint256 linkBalance_) {
+        linkBalance_ = s.link.balanceOf(address(this));
+    }
+
+    function vrfCoordinator() external view returns (address) {
+        return s.vrfCoordinator;
+    }
+
+    function link() external view returns (address) {
+        return address(s.link);
+    }
+
+    function keyHash() external view returns (bytes32) {
+        return s.keyHash;
+    }
+
+    function getMaxVrf() external view returns (uint256) {
+        return MAX_VRF;
+    }
+
+    function areGeodePrizesAvailable() public view returns (bool) {
```

Comment:
This function is called in openGeodes, which specifies geodeTokenIds. It might be better to also pass through a tokenIds arg to this function, otherwise I think it may return false for some situations where geode prizes are unavailable for some tokenIds but not others.

Reply:
this function and its usage is just meant to check whether there are any prizes left at all, not for any specific ids.